presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -36,8 +37,13 @@ private DependencyExtractor() {}
 
     public static Set<Symbol> extractUnique(PlanNode node)
     {
+        return extractUnique(node, n -> n);
+    }
+
+    public static Set<Symbol> extractUnique(PlanNode node, Lookup lookup)
```

Comment:
`DependencyExtractor.extractUnique(PlanNode node)` has not changed. It's underlying `ExpressionExtractor.extractExpressions(PlanNode plan)` neither. `ExpressionExtractor.Visitor` which is used in `extractExpressions` by default takes `Optional.empty()` in c-tor and the only change to the visitor is in `visitPlan` which only in case there's a `GroupReference` node visited, it resolves it using given `Lookup` and tries to visit the appropriate node (instead of `GroupReference` mock).
In other words:
1. If you don't use iterative plans, nothing has changed for you
2. If you use iterative plans, the only difference is you have to provide a `Lookup` and when there's a `GroupReference`, it's resolved to corresponding node, although there's no extra recursion added beyond what was there before (e.g. aforementioned `ExpressionExtractor.extractExpressionsNonRecursive`.

The default `extractUnique` didn't change the behavior.